Libs/MRML/Core/vtkMRMLNode.cxx

```diff
@@ -483,6 +483,9 @@ void vtkMRMLNode::ParseReferencesAttribute(const char* attValue, std::set<std::s
     // Only process this role if it has not been encountered already
     if (references.find(role) == references.end())
     {
+      // Clear all existing references for this role before adding new ones
+      this->RemoveNodeReferenceIDs(role.c_str());
```

Comment:
This is very simple, but could cause a lot of noise: it would invoke node reference removed + node reference added events for all the unchanged node references. Instead, we should compare existing and new references and only apply the difference.